homeassistant/components/xiaomi_aqara.py

```diff
@@ -201,20 +208,35 @@ class XiaomiDevice(Entity):
     def __init__(self, device, name, xiaomi_hub):
         """Initialize the Xiaomi device."""
         self._state = None
+        self._is_available = True
         self._sid = device['sid']
         self._name = '{}_{}'.format(name, self._sid)
         self._write_to_hub = xiaomi_hub.write_to_hub
         self._get_from_hub = xiaomi_hub.get_from_hub
         self._device_state_attributes = {}
-        xiaomi_hub.callbacks[self._sid].append(self.push_data)
-        self.parse_data(device['data'])
+        self._remove_unavailability_tracker = None
+        xiaomi_hub.callbacks[self._sid].append(self._add_push_data_job)
+        self.parse_data(device['data'], device['raw_data'])
         self.parse_voltage(device['data'])
 
+    def _add_push_data_job(self, *args):
+        self.hass.async_add_job(self.push_data, *args)
```

Comment:
Following the source, it looks like this callback will be called in a thread, so here we should use `self.hass.add_job`, and not the async version of the same method.

Reply:
Yes, it will be called in a thread. I will fix it in 5 hours.

Reply:
:+1: